src/components/Interface.ts

```diff
@@ -0,0 +1,40 @@
+import { Dispatch, SetStateAction } from 'react';
+
+interface IUserId {
+  userId: string;
+}
+
+interface IChatProps extends IUserId {
+  userId: string;
```

Comment:
IUserId를 상속하고 사용하지 않고 있습니다.

```suggestion
  userId: IUseerId;
```

위와 같이 상속받은 타입을 사용해서 작성하는것이 의도에 맞는것 같습니다!

+) Interface 라는 파일명 보다는 interfaces 를 사용하는것이 컨벤션에 맞는것 같고, type 관련 파일은 확장성을 고려하여 따로 디렉토리를 분리하는것이 좋을것 같습니다.